src/agent/src/process_options_unix.cpp

```diff
@@ -0,0 +1,95 @@
+#include <process_options.hpp>
+
+#include <agent.hpp>
+#include <agent_registration.hpp>
+#include <fmt/format.h>
+#include <fmt/ranges.h>
+#include <logger.hpp>
+#include <unix_daemon.hpp>
+
+#include <iostream>
+#include <vector>
+
+void RegisterAgent(const std::string& user,
+                   const std::string& password,
+                   const std::string& key,
+                   const std::string& name)
+{
+    if (!user.empty() && !password.empty() && !key.empty())
+    {
+        agent_registration::AgentRegistration reg(user, password, key, name);
+
+        http_client::HttpClient httpClient;
+        if (reg.Register(httpClient))
+        {
+            LogInfo("Agent registered.");
+        }
+        else
+        {
+            LogError("Registration fail.");
+        }
+    }
+    else
+    {
+        LogError("--user, --password and --key args are mandatory");
+    }
+}
+
+void StartAgent()
+{
```

Comment:
Wouldn't it be better to have an AgentDaemon::Start() function? As in the cases of StopAgent() and StatusAgent()?

Reply:
The idea behind AgentDaemon class is to deal with process daemonization, PIDs and such. So I'd rather limit the class' responsibility to that.